scripts/ensemble_kalman_paw_example.py

```diff
@@ -0,0 +1,251 @@
+        scaled_ensemble_stacks[k][:,0] = ensemble_stacks[k][:,0] - camera_means[0]
+        scaled_ensemble_stacks[k][:,1] = ensemble_stacks[k][:,1] - camera_means[1]
+        scaled_ensemble_stacks[k][:,2] = ensemble_stacks[k][:,2] - camera_means[2]
+        scaled_ensemble_stacks[k][:,3] = ensemble_stacks[k][:,3] - camera_means[3]
+    return scaled_ensemble_stacks
+
+def add_camera_means(ensemble_stacks, camera_means):
+    scaled_ensemble_stacks = ensemble_stacks.copy()
+    for k in range(len(ensemble_stacks)):
+        scaled_ensemble_stacks[k][:,0] = ensemble_stacks[k][:,0] + camera_means[0]
+        scaled_ensemble_stacks[k][:,1] = ensemble_stacks[k][:,1] + camera_means[1]
+        scaled_ensemble_stacks[k][:,2] = ensemble_stacks[k][:,2] + camera_means[2]
+        scaled_ensemble_stacks[k][:,3] = ensemble_stacks[k][:,3] + camera_means[3]
+    return scaled_ensemble_stacks
+
+def pca(S, n_comps):
+    pca_ = PCA(n_components=n_comps)
+    return pca_.fit(S), pca_.explained_variance_ratio_
+
+base_path = '/media/cat/cole/ibl-paw_ensembling/'
+video_name = '032ffcdf-7692-40b3-b9ff-8def1fc18b2e'
+paw = 'left'
+s = 2 #smoothing param
+quantile_keep_pca = 25
+
+keypoint_names = ['paw_l', 'paw_r']
+markers_list_left_cam = []
+markers_list_right_cam = []
+num_models = 10
+for i in range(num_models):
+    marker_path_left_cam = os.path.join(base_path, f'{video_name}.left.rng={i}.csv')
+    markers_tmp_left_cam = pd.read_csv(marker_path_left_cam, header=[0, 1, 2], index_col=0)
+    marker_path_right_cam = os.path.join(base_path, f'{video_name}.right.rng={i}.csv')
+    markers_tmp_right_cam = pd.read_csv(marker_path_right_cam, header=[0, 1, 2], index_col=0)
+    if '.dlc' not in marker_path_left_cam:
+        markers_tmp_left_cam = convert_lp_dlc(markers_tmp_left_cam, keypoint_names)
+    if '.dlc' not in marker_path_right_cam:
+        markers_tmp_right_cam = convert_lp_dlc(markers_tmp_right_cam, keypoint_names)
+    markers_list_left_cam.append(markers_tmp_left_cam)
+    #switch right camera paws
+    paw_l_x, paw_l_y, paw_l_likelihood = markers_tmp_right_cam['paw_l_x'].copy(), markers_tmp_right_cam['paw_l_y'].copy(), markers_tmp_right_cam['paw_l_likelihood'].copy()
+    paw_r_x, paw_r_y, paw_r_likelihood = markers_tmp_right_cam['paw_r_x'].copy(), markers_tmp_right_cam['paw_r_y'].copy(), markers_tmp_right_cam['paw_r_likelihood'].copy()
+    markers_tmp_right_cam['paw_l_x'], markers_tmp_right_cam['paw_l_y'], markers_tmp_right_cam['paw_l_likelihood'] = paw_r_x, paw_r_y, paw_r_likelihood
+    markers_tmp_right_cam['paw_r_x'], markers_tmp_right_cam['paw_r_y'], markers_tmp_right_cam['paw_r_likelihood'] = paw_l_x, paw_l_y, paw_l_likelihood
+    markers_list_right_cam.append(markers_tmp_right_cam)
+time_stamps_left_cam = np.load(os.path.join(base_path, f'{video_name}.timestamps.left.npy'))
+time_stamps_right_cam = np.load(os.path.join(base_path, f'{video_name}.timestamps.right.npy'))
+
+markers_list_stacked_interp = []
+markers_list_interp = [[],[]]
+
```

Comment:
add comment here about what the general idea of the next code block is (i.e. how the interpolation works)